cwltool/expression.py

```diff
@@ -53,14 +53,28 @@ def scanner(scan):  # type: (Text) -> List[int]
     SINGLE_QUOTE = 4
     DOUBLE_QUOTE = 5
     BACKSLASH = 6
+    COMMENT = 7
 
     i = 0
     stack = [DEFAULT]
     start = 0
+    com_check = ''
     while i < len(scan):
         state = stack[-1]
         c = scan[i]
 
+        if state == COMMENT:
+            if c == '\n':
+                stack.pop()
+                com_check = ''
+            pass
+        com_check += c
```

Comment:
error: Incompatible types in assignment (expression has type "unicode", variable has type "str")

Reply:
Unfortunately I don't know if I'll actually be able to become familiar enough with the tests as these next few weeks are looking a bit busy for me. Perhaps I can reopen this when I get some time to understand better the system